enumerable.rb

```diff
@@ -0,0 +1,193 @@
+    my_arr = []
+    my_each do |x|
+      my_arr << x if yield(x)
+    end
+    my_arr
+  end
+
+  def my_all?(ver = nil)
+    if block_given?
+      puts 'this method'
+      my_each do |x|
+        return false unless yield(x)
+      end
+    elsif ver.nil?
+      puts 'this method'
+      my_each do |x|
+        return false if x.nil? || x == false
+      end
+    elsif ver.is_a? Class
+      my_each do |x|
+        return false unless x.is_a? ver
+      end
+    elsif ver.is_a? Regexp
+      my_each do |x|
+        return false unless x.match?(ver)
+      end
+    else
+      my_each do |x|
+        return false unless x == ver
+      end
+    end
+    true
+  end
+
+  def my_any?(ver = nil)
+    if block_given?
+      my_each do |x|
+        return true if yield(x)
+      end
+    elsif ver.nil?
+      my_each do |x|
+        return true if !x.nil? && x != false
+      end
+    elsif ver.is_a? Class
+      my_each do |x|
+        return true if x.is_a? ver
+      end
+    elsif ver.is_a? Regexp
+      my_each do |x|
+        return true if x.match?(ver)
+      end
+    else
+      my_each do |x|
+        return true if x == ver
+      end
+    end
+    false
+  end
+
+  def my_none?(ver = nil)
+    if block_given?
+      puts 'this method'
+      my_each do |x|
+        return false if yield(x)
+      end
+    elsif ver.nil?
+      puts 'this method'
+      my_each do |x|
+        return false if x == true
+      end
+    elsif ver.is_a? Class
+      my_each do |x|
+        return false if x.is_a? ver
+      end
+    elsif ver.is_a? Regexp
+      my_each do |x|
+        return false if x.match?(ver)
+      end
+    else
+      my_each do |x|
+        return false if x == ver
+      end
+    end
+    true
+  end
+
+  def my_count(num = nil)
+    cons = 0
+    if block_given?
+      my_each do |x|
+        cons += 1 if yield(x)
+      end
+      cons
+    elsif !num.nil?
+      my_each do |x|
+        cons += 1 if num == x
+      end
+      cons
+    else
+      length
+    end
+  end
+
+  def my_map(proc_g = nil)
+    return to_enum(:map) unless block_given?
+
+    my_arr = []
+    if proc_g.nil?
+      my_each do |x|
+        my_arr << yield(x)
+      end
+    else
+      my_each do |x|
+        my_arr << proc_g.call(x)
+      end
+    end
+    my_arr
+  end
+
+  def my_inject(opposym = nil, sym = nil)
+    cons = 1
+    sum = 1
+    if opposym.nil? && sym.nil?
+      my_each do |x|
+        cons = yield(cons, x)
+        sum += x
+      end
+      return cons - 1 if cons == sum
+    elsif !opposym.nil? && block_given?
+      my_each do |x|
+        opposym = yield(opposym, x)
+      end
+    elsif !opposym.nil? && !sym.nil?
+      my_each do |x|
+        opposym = opposym.method(sym).call(x)
+      end
+    else
+      my_each_with_index do |x, i|
+        cons = if i.zero?
+                 x
+               else
+                 cons.method(opposym).call(x)
+               end
+      end
+      return cons
+    end
+    opposym
+  end
```

Comment:
- [ ] identical and returns the same thing as ruby's [`inject`](https://ruby-doc.org/core-2.6.4/Enumerable.html#method-i-inject). ([Screenshot](https://gitlab.com/microverse/guides/projects/requirements_screenshots/raw/master/images/ruby/advanced_building_blocks_enumerable/my_inject.png) from [Odin](https://www.theodinproject.com/courses/ruby-programming/lessons/advanced-building-blocks#assignment-2))
- [ ] Your method has an error that needs fixing;
![inject](https://user-images.githubusercontent.com/19773024/71502490-8413be00-2881-11ea-9f08-3fa652dd4d08.png)